### ISSUE TITLE
Top of bindings having the most of open issues

### ISSUE THREAD

**Issue by lolodomo** [openhab/openhab-addons#13706]
Here is the top of binding having at least 9 open issues (2022 November 12):

1. mqtt (39)
2. deconz (20)
3. homematic (18)
4. knx (15)
5. miio (14)
6. shelly / homekit (13)
7. telegram / jdbc (10)
8. chromecast / influxdb (9)

Maintainers of these bindings are encouraged to check these issues and close some of them if they are no more valid.

**Comment by lolodomo** [comment]
Maybe maintainers of the bindings could look at the issues to see if they all still relevant ?

- @FStolte @gerrieg @mdicke2s for homematic
- @kaikreuzer  for knx
- @marcelrv for miio
- @markus7017 for shelly
- @andylintner @yfre for homekit

**Comment by lolodomo** [comment]
Other bindings which are below the top 10 but with several issues: jsscripting amazonechocontrol astro avmfritz bluetooth goecharger http hue keba loxone mihome modbus openwebnet smartmeter somfytahoma tesla tr064 tradfri

**Comment by markus7017** [comment]
re Shelly:
I closed some issues from previous PRs
some more will be closed by the pending PR

**Comment by yfre** [comment]
regarding homekit - I could close some old and obsolete ticket, waiting for feedback for one more. remaining 6 are valid feature requests and bugs.

**Comment by mvalla** [comment]
what about feature requests that are valid but not implemented yet? I do not think is useful to close them, as they may get addressed in the future

**Comment by lolodomo** [comment]
> what about feature requests that are valid but not implemented yet? I do not think is useful to close them, as they may get addressed in the future

Yes but at the same time, if a feature request is there since several years and generate no developer interest, we could consider it is time to close it? Maybe after two years?

**Comment by mvalla** [comment]
honestly I see this as just a loss of information. 
What is the practical drawback of keeping these feature request issues?
In case it's decided to close them, is there any way to tag them to be able to recover them quickly? (e.g. tag "closed-stale" or something better)

**Comment by florian-h05** [comment]
Regarding JS Scripting: I am working on https://github.com/openhab/openhab-addons/issues/12577, I’ll also have a look at the other open issues.
FYI: I’ve already reduced the number of open issues to 6.

**Comment by lolodomo** [comment]
> honestly I see this as just a loss of information.
> What is the practical drawback of keeping these feature request issues?
> In case it's decided to close them, is there any way to tag them to be able to recover them quickly? (e.g. tag "closed-stale" or something better)

Ok, I think we can keep them. Anyway, I am not deciding.

**Comment by lolodomo** [comment]
Thank you for your efforts, I will update the top during the weekend.
homekit and jsscripting will leave this top and miio is also about to leave it.

**Comment by lolodomo** [comment]
Maybe @ZzetT could have a look for the telegram binding, @kaikreuzer for the chromecast binding, @lujop for the influxdb binding

**Comment by hmerk** [comment]
@lolodomo Great to see your effort here.
Please keep in mind that some of the listed bindings have moved to smarthome/j so might not get updates here anymore.
We really need to find a solution for that.
Either somebody can take the effort and import the changes back, like i did for the mail binding, or we should remove them here to not have 2 different versions with different codebase.

Perhaps this would be worth an AC discussion.
The one I started is stale.

**Comment by lolodomo** [comment]
> Please keep in mind that some of the listed bindings have moved to smarthome/j so might not get updates here anymore. 

Why such anarchy / nightmare for the final user was introduced ? :(
The marketplace was created to develop non official bindings or beta versions of existing bindings, not to make some alternatives to the official bindings, I think this is clear in the rules of the marketplace.

> We really need to find a solution for that. Either somebody can take the effort and import the changes back, like i did for the mail binding, or we should remove them here to not have 2 different versions with different codebase.

A solution has to be found, that is clear !

What are the bindings concerned ?

> Perhaps this would be worth an AC discussion.

Probably. This is limited to only few members ? I am not sure I have an access...

What disappointment for me to read that !

**Comment by J-N-K** [comment]
> > Please keep in mind that some of the listed bindings have moved to smarthome/j so might not get updates here anymore.
> 
> Why such anarchy / nightmare for the final user was introduced ? :( The marketplace was created to develop non official bindings or beta versions of existing bindings, not to make some alternatives to the official bindings, I think this is clear in the rules of the marketplace.
> 

This is not true. The rules have ben adjusted to this version AFTER the marketplace was introduced. Anyway, even without the marketplace (or any other add-on service): you can't force people to develop within the openHAB organization. There always have been add-ons not included in the distribution and there always have been modified versions of add-ons.

**Comment by florian-h05** [comment]
> you can't force people to develop within the openHAB organization. There always have been add-ons not included in the distribution and there always have been modified versions of add-ons.

I am wondering why one doesn't want to develop within the openHAB organization.

If there is a reason/problem, we maybe can discuss and solve it, because having forks of official add-ons is type of confusing for the end users.

**Comment by ZzetT** [comment]
Hi @lolodomo 
thanks for letting me know that there are several open issues for the telegram binding. I have to admit that I don't use openhab as my smarthome system any longer (for quite some time already). Thats why I lost a bit of interest to look at these issues.
Not sure if you can find any other lead developer for that. Otherwise I have to try to setup a new development environment first to try to work on that issues.
The binding itself uses the [java-telegram-bot-api](https://github.com/pengrad/java-telegram-bot-api) internally and only acts as a bridge between the openhab system and that Java library. So any feature that is supported by that library already, shouldn't be too difficult to add to the binding.
Sorry for that....

**Comment by lolodomo** [comment]
> I am wondering why one doesn't want to develop within the openHAB organization.

I am sure the following discussion will irritate me and disappoint me.
Can we please avoid to discuss about that in THAT topic.
The object of that topic is just to have a picture of bindings with issues/requests for enhancement and to motivate developers to check them.

**Comment by ccutrer** [comment]
> > I am wondering why one doesn't want to develop within the openHAB organization.
> 
> Can we please avoid to discuss about that in [THIS] topic.
> 
> The object of [this] topic is just to have a picture of bindings with lots of issues/requests for enhancement and to motivate maintainers to check them.

While I'm normally as annoyed by not focusing on the topic at hand, I think it could be relevant. Though I do want to focus my comments on resolving issues, and hopefully without offending any involved party, since that's not my intent. Anyhow, I think that having this discussion _is_ on topic, since resolving it will allow cleaning up  large chunks of issues, which is the core ask for THIS original topic.

I don't know the full history here, since I haven't been involved with openHAB for as long as others. But I _have_ worked with and committed to all three repositories (openhab-core, openhab-addons, and smarthomej/addons) within the last few months. I'd agree with @lolodomo that it can be confusing and frustrating to an end user when an official plugin has issues, and then you find out there's a different version that's more well maintained. On the flip side, I also agree with @J-N-K that it's impossible to prevent someone from maintaining an alternate version of a binding outside of this repository -- being able to easily fork, improve, and share your improvements in whatever manner is what makes open source great.

So how do you reconcile those two things? It seems there are a few choices:
1. The status quo. You don't. The official addons bitrot, and continue to ignore that smarthomej versions exist.
2. The official addons continue to bitrot, but at least add a note in their READMEs that acknowledge the existence of and links to smarthomej. Not a great experience, but at least makes it more possible for an end user to discover smarthomej, instead of getting frustrated at brokenness and apparent non-maintenance of an addon that might be critical to them, and losing them from the openHAB platform altogether.
3. Smarthomej (or at least those addons that duplicate official addons, but maybe all of them) merges back into the official repo. Sounds like there's history here that may make this difficult.
4. openhab-addons admits that their version of these bindings have been abandoned, and that better versions are available in the community. Just remove them completely, possibly at least leaving their readmes in place on the openHAB docs website pointing to smarthomej.

Putting myself in the shoes of an end user unfamiliar with the situation, option 3 would obviously be best. Option 4 also wouldn't be so bad - and would add even more weight to OpenHAB officially endorsing "there are community addons that we don't officially maintain or endorse individually, but _do_ endorse other community member's freedom to maintain and distribute addons that may not be fit to be in the main distribution." Option 1 definitely seems like the worst of the 4.

So... please! Can we at least move away from option 1?!

**Comment by florian-h05** [comment]
I agree that the status quo is not great. The confusion we put on the end users by having multiple forks of some addons, and, as ccutrer stated out, the risk that users are 

> getting frustrated at brokenness and apparent non-maintenance of an addon that might be critical to them, and losing them from the openHAB platform altogether.

shouldn‘t be accepted here. I am aware that there is some history that lead to forked addons, and I personally welcome that there are addons developed somewhere else, as this is open source, but I‘d be happy if come to a decision here (or in another thread).

Coming to the choices that ccutrer listed:

1. Please not this way
2. If don‘t come to a better solution, we should at least choose this way. FYI: HomeAssistant also mentions community forks of their integrations in their docs (see https://www.home-assistant.io/integrations/influxdb/).
3. The best option, as the addons then directly come with the openHAB distribution. But for some addons, the problem is the relatively long release cycle of openHAB. Those addons provide support for APIs with many breaking changes in short time periods. Anyway, for the addons that don‘t use such APIs, we‘d need some changes to the rules to make this happen.
4. Also an option, as it gives the user well-maintained add-ons and, as ccutrer stated, adds more weight to the community aspect of openHAB. If the community addons aren‘t maintained anymore at some point future, openHAB could still backport them into the official distribution to maintain them there, or don‘t maintain them, as this issue shows :-(

**Comment by lolodomo** [comment]
@wborn @kaikreuzer : do you know how to move all the messages about openHAB vs smarthomej elsewhere (separate issue?) to continue that interesting discussion not in this issue ?
I would like to keep the positive feeling in this issue with all the great maintainers that made the effort to look at open issues.

**Comment by lujop** [comment]
> 
I'm sorry, I've been not able to dedicate time to the addon, and in the short term, I don't expect that to change.
If some one can help with the issues will be heavily appreciated, if not I won't be able to attend to them at the moment.

**Comment by wborn** [comment]
> do you know how to move all the messages about openHAB vs smarthomej elsewhere (separate issue?) to continue that interesting discussion not in this issue ?

I don't think there is any built-in functionality for that in GitHub. The closest thing would probably be to split it off manually by creating a separate issue and then copy/paste/quote all comments from this issue to that issue and delete then here. E.g. by imitating the bot in https://github.com/openhab/openhab-core/issues/2276.

**Comment by Skinah** [comment]
Thanks for taking the time to compile the list and also all that you do here @lolodomo 
I can probably take a look at some of the telegram issues over the Christmas break.

Wanted to raise that some of the people marked as the code owner / maintainer may no longer be part of this project (unless this has been looked at recently), I suspect this is the case with telegram and also mqtt which feature in your top 10 list. Perhaps getting new people that make decent PR in recent times as the new codeowner may help reduce the workload.

**Comment by lsiepel** [comment]
Thanks @lolodomo for this list. Just created a new issue to proceed that discussion about the bindings developed elsewhere. So hopefully we can get back to this topic.  :-)

Do you have a simple tool to generate these statistics? Maybe worth spending a bit of time on the github api to automate a monthly report or per milestone release update of this top 10 list?

**Comment by lolodomo** [comment]
> Do you have a simple tool to generate these statistics? Maybe worth spending a bit of time on the github api to automate a monthly report or per milestone release update of this top 10 list?

No, it is done and updated manually. 
The initial search was time consuming, the updates are less (mainly clicking on each existing link). 
I will update it today for the new year.

**Comment by adisD** [comment]
@lolodomo I do really appreciate your initiative with highlighting erroneous add-ons. However it looks like not much happened regarding closing those. What it normally takes to fix #13692 like in my case deconz add-on. I must stay with OH 3.2 as none of my zigbee sensors (all my sensors are zigbee) get registered. best regards

**Comment by lolodomo** [comment]
> @lolodomo I do really appreciate your initiative with highlighting erroneous add-ons. However it looks like not much happened regarding closing those.

Maintainers already helped to reduce the number of issues/RFC for certain bindings, thank you to them.
I evolves of course less for the bindings for which we don't have any more active maintainers.

**Comment by lsiepel** [comment]
> 

Because there are 550+ issues and new ones appear while old ones get closed, it might not be very visible. But yeah manyu issues have been resolved, closed or validated. I'm still chasing the older issues and it seems many of htem have been fixed. 
You are right about deCONZ, that is a 'special' case, please check https://github.com/openhab/openhab-addons/issues/14124

**Comment by lsiepel** [comment]
> @lolodomo I do really appreciate your initiative with highlighting erroneous add-ons. However it looks like not much happened regarding closing those. What it normally takes to fix #13692 like in my case deconz add-on. I must stay with OH 3.2 as none of my zigbee sensors (all my sensors are zigbee) get registered. best regards

Because there are 550+ issues and new ones appear while old ones get closed, it might not be very visible. But yeah manyu issues have been resolved, closed or validated. I'm still chasing the older issues and it seems many of htem have been fixed. 
You are right about deCONZ, that is a 'special' case, please check https://github.com/openhab/openhab-addons/issues/14124

**Comment by andrewfg** [comment]
@lolodomo your statement that Hue has 10 open issues is actually a bit unfair; a couple of the issues refer to the Hue Emulation binding, and one refers to the Deconz binding. But nevertheless many thanks for the issue tracking.

**Comment by lolodomo** [comment]
> @lolodomo your statement that Hue has 10 open issues is actually a bit unfair; a couple of the issues refer to the Hue Emulation binding, and one refers to the Deconz binding. But nevertheless many thanks for the issue tracking.

No, I really counted only hue. We have now 11 for hue.

**Comment by adisD** [comment]
> > > Please keep in mind that some of the listed bindings have moved to smarthome/j so might not get updates here anymore.
> > 
> > 
> > Why such anarchy / nightmare for the final user was introduced ? :( The marketplace was created to develop non official bindings or beta versions of existing bindings, not to make some alternatives to the official bindings, I think this is clear in the rules of the marketplace.
> 
> This is not true. The rules have ben adjusted to this version AFTER the marketplace was introduced. Anyway, even without the marketplace (or any other add-on service): you can't force people to develop within the openHAB organization. There always have been add-ons not included in the distribution and there always have been modified versions of add-ons.

I am not new to OpenHAB although not engaged as a code maintainer. I am missing to see how spread of add-on install bundles helps to end users. I had to downgrade to OH 3.2 since "standard" deconz addon suddenly got broken. Recently learned the hard way that there is smarthome/j where solution may be. It still remains for me to find whether smarthome/j is really providing deconz add-on compatible with OH 3.4. What is smarthome/j? Another home automation platform?

**Comment by jlaur** [comment]
> Recently learned the hard way that there is smarthome/j where solution may be. It still remains for me to find whether smarthome/j is really providing deconz add-on compatible with OH 3.4. What is smarthome/j? Another home automation platform?

This topic has moved to #14129. See also #14124 which is specific to your case.

**Comment by lsiepel** [comment]
Created a very basic script that can help automate the list creation. It is based on the Github API. Just insert your personal API KEY and it logs the issue count per binding. 

https://playcode.io/1245173

Let me know if you think it can be of any help and/or if we can improve it. I hope this can add some value to reducing the issue count / bugs / enhancements.

**Comment by lsiepel** [comment]
TOP 40: 

01. mqtt 22 
02. shelly 21 
03. openwebnet 17 
04. knx 16 
05. homematic 15 
06. homekit 14 
07. miio 11 
08. telegram 10 
09. http 10 
10. netatmo 10 
11. avmfritz 9 
12. mihome 8 
13. influxdb 8 
14. hue 8 
15. gpio 8 
16. tradfri 7 
17. amazonechocontrol 7 
18. tesla 7 
19. somfytahoma 7 
20. deconz 7 
21. smartmeter 6 
22. loxone 6 
23. hueemulation 6 
24. enocean 6 
25. jdbc 6 
26. chromecast 6 
27. paradoxalarm 6 
28. samsungtv 5 
29. rrd4j 5 
30. tacmi 5 
31. modbus 5 
32. keba 5 
33. ipcamera 5 
34. goecharger 5 
35. gardena 4 
36. openhabcloud 4 
37. kostalinverter 4 
38. icalendar 4 
39. daikin 4 
40. nibeheatpump 4

**Comment by marcelrv** [comment]
@lsiepel nice job.
seeing the count for miio binding, seems there is dome double counting or something. 
At least when I manually count for miio I come to 7 as opposed to the 11 in the list

**Comment by lsiepel** [comment]
Ah, it also includes pull requests. I have updated the script to filter those out of the query. And altered it a bit so it generated markup that can be copy/paste here immediate with url to details. 

List of all addons  with 5 or more issues:

1. [mqtt](https://github.com/openhab/openhab-addons/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+mqtt) 20
2. [shelly](https://github.com/openhab/openhab-addons/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+shelly) 19
3. [homematic](https://github.com/openhab/openhab-addons/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+homematic) 15
4. [openwebnet](https://github.com/openhab/openhab-addons/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+openwebnet) 15
5. [knx](https://github.com/openhab/openhab-addons/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+knx) 13
6. [homekit](https://github.com/openhab/openhab-addons/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+homekit) 13
7. [telegram](https://github.com/openhab/openhab-addons/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+telegram) 10
8. [http](https://github.com/openhab/openhab-addons/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+http) 10
9. [mihome](https://github.com/openhab/openhab-addons/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+mihome) 8
10. [tradfri](https://github.com/openhab/openhab-addons/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+tradfri) 7
11. [hue](https://github.com/openhab/openhab-addons/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+hue) 7
12. [amazonechocontrol](https://github.com/openhab/openhab-addons/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+amazonechocontrol) 7
13. [tesla](https://github.com/openhab/openhab-addons/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+tesla) 7
14. [somfytahoma](https://github.com/openhab/openhab-addons/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+somfytahoma) 7
15. [netatmo](https://github.com/openhab/openhab-addons/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+netatmo) 7
16. [influxdb](https://github.com/openhab/openhab-addons/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+influxdb) 6
17. [smartmeter](https://github.com/openhab/openhab-addons/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+smartmeter) 6
18. [loxone](https://github.com/openhab/openhab-addons/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+loxone) 6
19. [hueemulation](https://github.com/openhab/openhab-addons/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+hueemulation) 6
20. [jdbc](https://github.com/openhab/openhab-addons/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+jdbc) 6
21. [avmfritz](https://github.com/openhab/openhab-addons/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+avmfritz) 6
22. [gpio](https://github.com/openhab/openhab-addons/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+gpio) 6
23. [miio](https://github.com/openhab/openhab-addons/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+miio) 5
24. [deconz](https://github.com/openhab/openhab-addons/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+deconz) 5
25. [rrd4j](https://github.com/openhab/openhab-addons/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+rrd4j) 5
26. [chromecast](https://github.com/openhab/openhab-addons/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+chromecast) 5

**Comment by andrewfg** [comment]
> it also includes pull requests

And it includes issues where ther is a pull request already available to solve the issue, but nobody is willing to review the pull request, and thus solve the issue specifically, (and probably other issues too)

**Comment by lsiepel** [comment]
> > it also includes pull requests
> 
> And it includes issues where there is a pull request already available to solve the issue, but nobody is willing to review the pull request, and thus solve the issue specifically, (and probably solve other issues too)

Don’t think it is a matter of not willing to review. It’s rather a lack of time. But that’s somewhat off topic. The list consists of open issues and is meant to engage and solve issues. Let me know if there are issues or improvements needed to the supplied snippet.

**Comment by andrewfg** [comment]
^
My point is that it is a waste of time trying to engage binding developers, if reviewers are not engaged to receive their efforts. IMHO.

**Comment by lolodomo** [comment]
Here are all the commits since one year (https://github.com/openhab/openhab-addons/graphs/commit-activity)
![image](https://user-images.githubusercontent.com/10186704/227699736-6a95a1cb-78b5-4875-893b-a6e2b01e1580.png)
297 since the beginning of 2023.

37 Pull requests merged the last week. https://github.com/openhab/openhab-addons/pulse
106 Pull requests merged the last month. https://github.com/openhab/openhab-addons/pulse/monthly

**Comment by markus7017** [comment]
I updated / fixed most of the open issues, they will be closed with the next PR merge.

**Comment by J-N-K** [comment]
@lolodomo Can you update the list?

**Comment by lsiepel** [comment]
[shelly](https://github.com/openhab/openhab-addons/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+shelly) 22
[mqtt](https://github.com/openhab/openhab-addons/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+mqtt) 21
[homematic](https://github.com/openhab/openhab-addons/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+homematic) 18
[homekit](https://github.com/openhab/openhab-addons/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+homekit) 13
[openwebnet](https://github.com/openhab/openhab-addons/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+openwebnet) 11
[telegram](https://github.com/openhab/openhab-addons/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+telegram) 10
[http](https://github.com/openhab/openhab-addons/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+http) 10
[knx](https://github.com/openhab/openhab-addons/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+knx) 9
[tesla](https://github.com/openhab/openhab-addons/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+tesla) 9
[mihome](https://github.com/openhab/openhab-addons/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+mihome) 8
[jdbc](https://github.com/openhab/openhab-addons/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+jdbc) 8
[hue](https://github.com/openhab/openhab-addons/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+hue) 7
[amazonechocontrol](https://github.com/openhab/openhab-addons/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+amazonechocontrol) 7
[tradfri](https://github.com/openhab/openhab-addons/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+tradfri) 7
[smartmeter](https://github.com/openhab/openhab-addons/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+smartmeter) 7
[ipcamera](https://github.com/openhab/openhab-addons/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+ipcamera) 7
[loxone](https://github.com/openhab/openhab-addons/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+loxone) 6
[hueemulation](https://github.com/openhab/openhab-addons/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+hueemulation) 6
[gpio](https://github.com/openhab/openhab-addons/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+gpio) 6
[somfytahoma](https://github.com/openhab/openhab-addons/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+somfytahoma) 6
[avmfritz](https://github.com/openhab/openhab-addons/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+avmfritz) 5
[miio](https://github.com/openhab/openhab-addons/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+miio) 5
[rrd4j](https://github.com/openhab/openhab-addons/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+rrd4j) 5
[chromecast](https://github.com/openhab/openhab-addons/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+chromecast) 5
[netatmo](https://github.com/openhab/openhab-addons/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+netatmo) 5
[gardena](https://github.com/openhab/openhab-addons/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+gardena) 5
[velbus](https://github.com/openhab/openhab-addons/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+velbus) 5
[goecharger](https://github.com/openhab/openhab-addons/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+goecharger) 5

**Comment by Skinah** [comment]
Any chance that the MQTT issues can be sorted further? Separate into 
mqtt.homie
mqtt.espmilighthub
mqtt.homeassistant
etc....

**Comment by lolodomo** [comment]
Updated top:

1. [mqtt](https://github.com/openhab/openhab-addons/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+mqtt) (43) 
2. [homematic](https://github.com/openhab/openhab-addons/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+homematic) (18)
3. [homekit](https://github.com/openhab/openhab-addons/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+homekit) (13)
4. [telegram](https://github.com/openhab/openhab-addons/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+telegram) (11)
5. [http](https://github.com/openhab/openhab-addons/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+http) / [jdbc](https://github.com/openhab/openhab-addons/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+jdbc) / [tesla](https://github.com/openhab/openhab-addons/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+tesla) (10)
6. [knx](https://github.com/openhab/openhab-addons/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+knx) / [shelly](https://github.com/openhab/openhab-addons/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+shelly) (9)
7. [mihome](https://github.com/openhab/openhab-addons/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+mihome) / [tradfri](https://github.com/openhab/openhab-addons/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+tradfri) (8)

**Comment by Skinah** [comment]
Guessing this was accidentally unpinned? Have done it myself when scrolling on a mobile's tiny screen.

**Comment by J-N-K** [comment]
@lolodomo The http binding has only 4 issues, the other issues are related to other bindings. Maybe you need to grep for `[http]` instead of `http`.

**Comment by lsiepel** [comment]
[shelly](https://github.com/openhab/openhab-addons/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+shelly) 29
[homekit](https://github.com/openhab/openhab-addons/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+homekit) 20
[mqtt](https://github.com/openhab/openhab-addons/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+mqtt) 19
[homematic](https://github.com/openhab/openhab-addons/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+homematic) 17
[telegram](https://github.com/openhab/openhab-addons/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+telegram) 12
[tesla](https://github.com/openhab/openhab-addons/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+tesla) 9
[jdbc](https://github.com/openhab/openhab-addons/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+jdbc) 8
[amazonechocontrol](https://github.com/openhab/openhab-addons/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+amazonechocontrol) 8
[tradfri](https://github.com/openhab/openhab-addons/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+tradfri) 8
[sonos](https://github.com/openhab/openhab-addons/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+sonos) 7
[rrd4j](https://github.com/openhab/openhab-addons/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+rrd4j) 7
[openwebnet](https://github.com/openhab/openhab-addons/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+openwebnet) 7
[ecovacs](https://github.com/openhab/openhab-addons/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+ecovacs) 7
[avmfritz](https://github.com/openhab/openhab-addons/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+avmfritz) 7
[somfytahoma](https://github.com/openhab/openhab-addons/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+somfytahoma) 6
[velbus](https://github.com/openhab/openhab-addons/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+velbus) 6
[netatmo](https://github.com/openhab/openhab-addons/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+netatmo) 6
[mihome](https://github.com/openhab/openhab-addons/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+mihome) 5
[hueemulation](https://github.com/openhab/openhab-addons/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+hueemulation) 5
[knx](https://github.com/openhab/openhab-addons/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+knx) 5
[gpio](https://github.com/openhab/openhab-addons/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+gpio) 5
[kostalinverter](https://github.com/openhab/openhab-addons/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+kostalinverter) 5
[smaenergymeter](https://github.com/openhab/openhab-addons/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+smaenergymeter) 5
[dsmr](https://github.com/openhab/openhab-addons/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+dsmr) 5
[unifi](https://github.com/openhab/openhab-addons/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+unifi) 5
[modbus](https://github.com/openhab/openhab-addons/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+modbus) 5
[miio](https://github.com/openhab/openhab-addons/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+miio) 5
[icalendar](https://github.com/openhab/openhab-addons/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+icalendar) 5
[jsscripting](https://github.com/openhab/openhab-addons/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+jsscripting) 5
[boschshc](https://github.com/openhab/openhab-addons/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+boschshc) 5

Used the shared playcode snippet from before to generate this

**Comment by lsiepel** [comment]
List as of 25-11-2024: 
Shelly really needs some work i guess. 
[shelly](https://github.com/openhab/openhab-addons/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+shelly) 30
[homematic](https://github.com/openhab/openhab-addons/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+homematic) 14
[mqtt](https://github.com/openhab/openhab-addons/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+mqtt) 11
[telegram](https://github.com/openhab/openhab-addons/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+telegram) 11
[homekit](https://github.com/openhab/openhab-addons/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+homekit) 11
[amazonechocontrol](https://github.com/openhab/openhab-addons/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+amazonechocontrol) 10
[tradfri](https://github.com/openhab/openhab-addons/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+tradfri) 8
[avmfritz](https://github.com/openhab/openhab-addons/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+avmfritz) 8
[rrd4j](https://github.com/openhab/openhab-addons/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+rrd4j) 7
[jsscripting](https://github.com/openhab/openhab-addons/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+jsscripting) 7
[modbus](https://github.com/openhab/openhab-addons/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+modbus) 7
[enocean](https://github.com/openhab/openhab-addons/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+enocean) 7
[dsmr](https://github.com/openhab/openhab-addons/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+dsmr) 6
[somfytahoma](https://github.com/openhab/openhab-addons/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+somfytahoma) 6
[hue](https://github.com/openhab/openhab-addons/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+hue) 6
[tesla](https://github.com/openhab/openhab-addons/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+tesla) 6
[kostalinverter](https://github.com/openhab/openhab-addons/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+kostalinverter) 6
[miio](https://github.com/openhab/openhab-addons/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+miio) 6
[hueemulation](https://github.com/openhab/openhab-addons/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+hueemulation) 5
[jdbc](https://github.com/openhab/openhab-addons/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+jdbc) 5
[luxtronikheatpump](https://github.com/openhab/openhab-addons/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+luxtronikheatpump) 5
[network](https://github.com/openhab/openhab-addons/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+network) 5
[http](https://github.com/openhab/openhab-addons/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+http) 5
[sonos](https://github.com/openhab/openhab-addons/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+sonos) 5
[mihome](https://github.com/openhab/openhab-addons/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+mihome) 4
[loxone](https://github.com/openhab/openhab-addons/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+loxone) 4
[knx](https://github.com/openhab/openhab-addons/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+knx) 4
[rfxcom](https://github.com/openhab/openhab-addons/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+rfxcom) 4
[icalendar](https://github.com/openhab/openhab-addons/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+icalendar) 4
[freeathome](https://github.com/openhab/openhab-addons/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+freeathome) 4
[deconz](https://github.com/openhab/openhab-addons/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+deconz) 4
[yeelight](https://github.com/openhab/openhab-addons/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+yeelight) 4

**Comment by andrewfg** [comment]
Apropos Hue binding .. just so you know .. none of the mentioned issues are actually actionable (just read them yourself please) and I am therefore definitely not planning to take any actions.

**Comment by hmerk** [comment]
Same applies to many of the Shelly issues. Nearly half of them are enhancements request and a couple of bugs are awaiting feedback.

**Comment by lolodomo** [comment]
I am counting 40 opened "issues" for the Shelly binding .Are they really still opened ?
@markus7017 : maybe you could have a look?
Maybe also @jlaur as you seem more and more involved in the development of this binding?

**Comment by jlaur** [comment]
> Maybe also [@jlaur](https://github.com/jlaur) as you seem more and more involved in the development of this binding?

As you might [recognize](https://github.com/openhab/openhab-addons/pulls?q=is%3Apr+shelly+in%3Atitle+author%3Ajlaur), my focus has been to help getting the binding in a better shape by adding test coverage and refactoring some error-prone parts of the binding which historically were source of regressions. As part of this effort, I have also helped reviewing pull requests with the same focus and motivation.

I have only one device (PM Mini), which I still have not installed, so I'm not (yet?) using the binding on my production system.

I prefer to improve one thing at a time, so that means that I'm blocked while waiting for reviews. There is still room for improvements in many areas. Currently I'm stuck with #19229, and I don't even remember my next small "project", so I don't expect to be able to do much in current release cycle.

I sincerely hope that some of my contributions of reducing technical debt will make it easier to fix bugs without introducing new bugs, but I unfortunately can't commit to systematically start fixing our backlog of issues. I hope you understand.

**Comment by jlaur** [comment]
@lolodomo - do you have a script for generating your top 10?

**Comment by lolodomo** [comment]
> [@lolodomo](https://github.com/lolodomo) - do you have a script for generating your top 10?

No but @lsiepel has something I believe

**Comment by lsiepel** [comment]
Yes’ll update later today. I’ll update the start post to have a clear start post

**Comment by jlaur** [comment]
@lsiepel - is your script considering the issue labels "bug" or "enhancement"? In this case you might want to update it to consider issue type instead, since these labels will not be provided going forward (after #21110).

**Comment by lsiepel** [comment]
Ah yes, Will update the script and also update the results in the first post!

**Comment by lsiepel** [comment]
Updated the list in the startpost.
Also did some script improvements so the links are better filtering the issues. Not affected by the labels as all issues are treated equally.
I could change it to only list the issues without a linked pull request. Would that be usefull?

**Comment by andrewfg** [comment]
Minor point but 'hueemulation' is a different binding than 'hue' ..

**Comment by lsiepel** [comment]
What makes you say that?
They are both listed (one has 10, the other has 4 issues). It might happen that when you click the hue link, you also get a few hueemulation issues, they are not counted in the top 10, but cannot be filtered in the link. as the github filtering is crippled.

**Comment by andrewfg** [comment]
> when you click the hue link, you also get a few hueemulation issues

Indeed.
